tests/btf.cpp

```diff
@@ -0,0 +1,529 @@
+#include <fstream>
+#include <string>
+#include <unordered_set>
+
+#include "btf/btf.h"
+#include "btf/helpers.h"
+#include "gtest/gtest.h"
+
+namespace bpftrace::test::btf {
+
+using namespace bpftrace::btf;
+
+static std::string to_str(const AnyType &type)
+{
+  std::stringstream out;
+  out << type;
+  return out.str();
+}
+
+TEST(btf, to_str)
+{
+  Types btf;
+  auto int8 = btf.add<Integer>("int8", 1, 1);
+  ASSERT_TRUE(bool(int8));
+  EXPECT_EQ(to_str(AnyType(*int8)), "int8");
+}
+
+TEST(btf, kernel_types)
+{
+  std::ifstream file("/sys/kernel/btf/vmlinux");
```

Comment:
Is this file always available everywhere this test might get run? What about just vendoring part of it for this test?

Reply:
Will do.